src/sortingtraining/Strategy.java

```diff
@@ -0,0 +1,11 @@
+package sortingtraining;
+
+public enum Strategy {
```

Comment:
Having `enum`, let's take advantage of that.
It can have private variable type of `Strategy`. You'll be forced to initialize it, and it can do as much as `SortingStrategySwitchResolver`: return different strategy for each `enum` value. Please refer to this stack [for snippet ](https://stackoverflow.com/a/9825939)of what I had in mind